index.js

```diff
@@ -1039,6 +1039,78 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  //# takeWhile :: (Foldable f, Monoid (f a)) =>  (a -> Boolean) -> f a -> f a
+  //.
+  //. Returns the first elements for which the predicate returns true.
+  //.
+  //. ```javascript
+  //. > S.takeWhile(c => c < 'c', ['a', 'b', 'c', 'd', 'e'])
+  //. ['a', 'b']
+  //.
+  //. > S.takeWhile(c => c < 'a', ['a', 'b', 'c'])
+  //. []
+  //.
+  //. > S.takeWhile(S.K(true), S.Just(1))
+  //. Just(1)
+  //.
+  //. > S.takeWhile(S.K(false), S.Just(1))
+  //. Nothing
```

Comment:
I was trying to show an example of using a non-array. If we're not allowing strings, I can't think of a good example.